src/azure-cli-core/azure/cli/core/commands/__init__.py

```diff
@@ -366,6 +372,25 @@ def _build_kwargs(self, func, ns):  # pylint: disable=no-self-use
             kwargs['ns'] = ns
         return kwargs
 
+    def _rudimentary_get_command(self, args):  # pylint: disable=no-self-use
+        """ Rudimentary parsing to get the command """
+        nouns = []
+        command_names = self.commands_loader.command_table.keys()
+        for arg in args:
+            if arg[0] != '-':
+                nouns.append(arg)
+            else:
+                break
+
+        def _find_args(args):
+            search = ' '.join(args)
+            return next((x for x in command_names if x.startswith(search)), False)
```

Comment:
Is this slow?
There could be lots of commands.

Reply:
Slower than the old method sure, but the performance test script still passes without much change.

Reply:
I'll see if I can just index the dictionary. I think I tried that and it had unintended consequences.